nue-zigbee-smart-fan-light-switch.groovy

```diff
@@ -78,237 +74,230 @@ void initialize() {
+
+        if (event.name == "switch") {
+            sendChildEvent("${device.deviceNetworkId}-Light", "switch", event.value)
+        } else if (event.name == "fan") {
+            sendChildEvent("${device.deviceNetworkId}-Fan", "speed", event.value)
+        }
     }
-    
 }
 
 /* ======== capability commands ======== */
 
 def refresh() {
     if (enableDebug) log.debug "refresh()"
-   
-    return zigbee.onOffRefresh() + zigbee.readAttribute(0x0202, 0x0000)
+
+    return zigbee.onOffRefresh() + zigbee.readAttribute(0x0006, 0x0000)
 }
 
 
 def List<String> configure() {
     if (enableDebug) log.debug "configure()"
-    
+
     // clean stuff up !!
     state.clear()
-    
-    getChildDevices().each { 
+
+    getChildDevices().each {
         if (enableDebug) log.debug "deleteChildDevice ${it.deviceNetworkId}"
-        deleteChildDevice("${it.deviceNetworkId}") 
+        deleteChildDevice("${it.deviceNetworkId}")
     }
-    
-    addChildType("Light","Switch")
-    addChildType("Fan","Fan")
-    state.comments = "Child devices created!" 
-    return zigbee.configureReporting(0x0006, 0x0000, 0x10, 1, 600, null) +
-       zigbee.configureReporting(0x0202, 0x0000, 0x30, 1, 600, null)
+
+    addChildType("Light", "Light")
+    addChildType("Fan", "Fan")
+    state.comments = "Child devices created!"
+    return zigbee.configureReporting(0x0006, 0x0000, 0x10, 1, 600, null)
 }
 
 /* ======== custom commands and methods ======== */
 
-def calcChildFanSpeed() {
-    if (enableDebug) log.debug "calcChildFanSpeed()"
-    
-    def fanspeed = "off"
-    def fanswitch = "off"
-    
-    
-    if (device.currentValue("fan") == "high") {
-        fanspeed = "high"
-        fanswitch = "on"
-    }
-    if (device.currentValue("fan") == "medium") {
-        fanspeed = "medium"
-        fanswitch = "on"
-    }
-    if (device.currentValue("fan") == "low") {
-        fanspeed = "low"
-        fanswitch = "on"
-    }
-    getChildDeviceParse("${device.deviceNetworkId}-Fan","speed",fanspeed) 
-    getChildDeviceParse("${device.deviceNetworkId}-Fan","switch",fanswitch) 
-    getChildDeviceParse("${device.deviceNetworkId}-Fan","supportedFanSpeeds",["low", "medium", "high", "off"])
-}
-
 def addChildType(String label, String type) {
-    if (enableDebug) log.debug "addChildType(${label},${type})" 
+    if (enableDebug) log.debug "addChildType(${label},${type})"
     def devLabel = device.label
     if (devLabel == null) devLabel = device.name
-    
+
     switch (type) {
-        case "Switch":
+        case "Light":
             addChildDevice("hubitat","Generic Component Switch", "${device.deviceNetworkId}-${label}", [label: "${devLabel}-${label}",name: "${device.name}-${label}", isComponent: true])
             break
-        
+
         case "Fan":
             addChildDevice("hubitat","Generic Component Fan Control", "${device.deviceNetworkId}-${label}", [label: "${devLabel}-${label}",name: "${device.name}-${label}", isComponent: true])
```

Comment:
FYI I was able to get the child fan component to show the limited list of options by creating a custom component and applying the constraint there. (Calling to "command" etc in the way the parent component sets the constraint doesn't work). Although it looks like the dashboard fan template also uses the expanded list and does not try pulling in the constraint from the device so the point is somewhat moot.. 

Basic device handler looks like this:
```
import groovy.transform.Field

@Field static final List supportedFanSpeeds = ["low", "medium", "high", "off"]

metadata {
    definition (name: "Basic Component Fan Control", namespace: "abrom", author: "Andrew Bromwich") {
        capability "FanControl"
        capability "Refresh"

        command "setSpeed", [[name: "Fan speed*", type: "ENUM", description: "Fan speed to set", constraints: supportedFanSpeeds]]
    }
}

String cycleSpeed(cd) {
    getParent().cycleSpeed()
}

String setSpeed(speed) {
    getParent().setSpeed(speed)
}
```

Then of course just change this line to:
```
addChildDevice("abrom","Basic Component Fan Control", "${device.deviceNetworkId}-${label}", [label: "${devLabel}-${label}",name: "${device.name}-${label}", isComponent: true])